src-java/kilda-persistence-orientdb/src/main/java/org/openkilda/persistence/orientdb/repositories/OrientDbFlowEventRepository.java

```diff
@@ -44,4 +55,69 @@ public boolean existsByTaskId(String taskId) {
             return results.iterator().hasNext();
         }
     }
+
+    @Override
+    public Map<Instant, String> findFlowStatusesByFlowIdAndTimeFrame(String flowId,
+                                                                     Instant timeFrom, Instant timeTo, int maxCount) {
+        Map<Instant, String> statuses = new TreeMap<>();
```

Comment:
does it make sense to replace map to sortedmap here, to highlight the nature of the data?

Reply:
This was done for clarity, to see the sequence of status changes.

Reply:
Now `SortedMap` is used instead of `Map`